src/interceptors/http.ts

```diff
@@ -0,0 +1,120 @@
+import http, { RequestOptions, ClientRequest, IncomingMessage } from 'http'
+import decompressResponse from 'decompress-response'
+import pick from 'lodash/pick'
+import crypto from 'crypto'
+import { URL } from 'url'
+
+type Callback = (res: IncomingMessage) => void
+class ExpressoHttpInterceptor {
+    data: Array<any> = []
+
+    constructor() {
+        const old = http.request
+        http.request = (
+            url: string | RequestOptions | URL,
+            options: RequestOptions | Callback | undefined,
+            callback?: Callback
+        ): ClientRequest => {
+            const startAt = process.hrtime()
+            let request
+            if (this.isURL(url)) {
+                const opt = options as RequestOptions
+                const cb = this.overrideCallback(opt, callback as Callback, startAt)
```

Comment:
on pourrait sortir du `if`: `const cb = this.overrideCallback(opt, options as Callback, startAt)` comme on le fait dans le else également

Reply:
Done bien vu.

Reply:
En fait on peu pas la veluer de opt n'est pas la même